### PR TITLE
[fr] #4091

### PR DESCRIPTION
**Pull request by cake17**
@antograssiot @HavokInspiration I'm not sur about some translations about events, if you have corrections to suggest ?

### CODE REVIEW

File: fr/core-libraries/events.rst [hunk 1]
```diff
@@ -131,17 +131,42 @@ gestionnaire global entraîne une complexité supplémentaire à gérer.
...
 objets séparés et même le déléguer à des tâches offline si vous avez ce
 besoin.
 
+.. _tracking-events:
+
+Suivre la Trace des Événements
+------------------------------
+
+Pour garder une liste des événements qui sont déclenchés pour un
+``EventManager`` en particulier, vous pouvez activer le tracking d'événement.
```
Comment (HavokInspiration) [inline fr/core-libraries/events.rst]: événements au pluriel ?

File: fr/core-libraries/events.rst [hunk 2]
```diff
@@ -131,17 +131,42 @@ gestionnaire global entraîne une complexité supplémentaire à gérer.
...
+à partir de la liste d'événements::
+
+    $eventsFired = EventManager::instance()->getEventList();
+    $firstEvent = $eventsFired[0];
+
+Le tracking peut être désactivé en retirant la liste d'événements ou en appelant
+:php:meth:`Cake\\Event\\EventList::trackEvents(false)`
+
+.. versionadded:: 3.2.10
+    Le tracking d'événement et :php:class:`Cake\\Event\\EventList` ont été
```
Comment (HavokInspiration) [inline fr/core-libraries/events.rst]: événements au pluriel ?

**Comment by HavokInspiration** [comment]
I didn't see it was a big diff, I don't have time to finish it right now. I stopped [there](https://github.com/cakephp/docs/pull/4092/files#diff-2225b783830f5bf83e18190b7e4dce72L231) and it looks good to me up to there 😄

**Comment by cake17** [comment]
ok 😄 no problem, thanks for the review

File: fr/core-libraries/events.rst [hunk 3]
```diff
@@ -220,38 +245,40 @@ que PHP supporte::
...
-event listeners on the default events. Lets take an example  'UserFeedback'
-plugin which handles feedback forms from users. From your application you would
-like to know when a Feedback record has been saved and ultimately act on it. You
-can could listen to the global ``Model.afterSave`` event.  However, you can take
-a more direct approach and only listen to the event you really need::
-
-    // You can create the following before the
-    // save operation, ie. config/bootstrap.php
+Quand vous travaillez avec des plugins qui ne déclenchent pas d'événement
+spécifique, vous pouvez utiliser les listeners d'événement sur les événements
```
Comment (antograssiot) [inline fr/core-libraries/events.rst]: les listeners d'événements ?

File: fr/development/testing.rst [hunk 4]
```diff
@@ -1442,6 +1442,110 @@ un message indiquant 1 passé et 4 assertions.
...
+
+        public function removeFromCart(Event $event)
+        {
+            $order = $event->data('order');
+            $this->delete($order->cart_id);
+        }
+    }
+
+.. note::
+    Pour faire des asserts sur le fait que des événements sont déclenchés, vous
```
Comment (antograssiot) [inline fr/development/testing.rst]: Pour faire des assertions ?

File: fr/development/testing.rst [hunk 5]
```diff
@@ -1442,6 +1442,110 @@ un message indiquant 1 passé et 4 assertions.
...
+    devez d'abord activer :ref:`tracking-events` sur le gestionnaire
+    d'événements pour lequel vous souhaitez faire des asserts.
```
Comment (antograssiot) [inline fr/development/testing.rst]: idem

**Comment by antograssiot** [comment]
nice work @cake17 as usual

File: fr/development/testing.rst [hunk 6]
```diff
@@ -1442,6 +1442,110 @@ un message indiquant 1 passé et 4 assertions.
...
+
+Pour tester le ``OrdersTable`` du dessus, nous activons le tracking dans
+``setUp`` puis faisons des asserts que l'événement a été déclenché, et faisons
+un assert que l'entity ``$order`` a été passée dans les données de l'événement::
+
```
Comment (antograssiot) [inline fr/development/testing.rst]: The original text could be updated also to remove the use of "we" in this sentence in favor of "you" like in most places in the book

**Comment by cake17** [comment]
@antograssiot changes done, this PR can be merged, thanks for the review 🍺

**Pull request merged**